conduit/pipeline/pipeline.go

```diff
@@ -372,7 +398,7 @@ func (p *pipelineImpl) Init() error {
 }
 
 func (p *pipelineImpl) Stop() {
-	p.cf()
+	p.ccf(stopCause)
```

Comment:
This is a good point. If the only way cancel under non-error conditions is through `Stop()` or we don't need to disambiguate, we should use `nil` here. On the other hand, if we have other ways to _happy-path-cancel_ the context, and want to tell them apart, then we'd need this approach. I don't have a strong opinion here.

Reply:
I feel like you've looked into graceful pipeline exits before/whether we properly respond to things like sigterm by running some shutdown logic. In that case I would expect to attempt to call Stop and for it to return without error right?

Reply:
It's a good juncture to experiment a bit and see how to get the graceful exit to work properly, and this will hopefully lead the way to the simplest cancellation approach. Some references:
* #100 
* #97 
*

Reply:
Let's continue this discussion on [this thread](https://github.com/algorand/conduit/pull/128#discussion_r1289543345)